vendor/engines/sanger_sequencing/spec/models/sanger_sequencing/batch_spec.rb

```diff
@@ -0,0 +1,27 @@
+require "rails_helper"
+
+RSpec.describe SangerSequencing::Batch do
+  describe ".for_product_group" do
+    subject { described_class.for_product_group(group) }
+
+    before { FactoryGirl.create(:product_group, :fragment) }
+    let!(:blank_batch) { FactoryGirl.create(:sanger_sequencing_batch, group: "") }
+    let!(:nil_batch) { FactoryGirl.create(:sanger_sequencing_batch, group: nil) }
+    let!(:grouped_batch) { FactoryGirl.create(:sanger_sequencing_batch, group: "fragment") }
+
+    context "with group given as nil" do
+      let(:group) { nil }
+      it { is_expected.to eq [blank_batch, nil_batch] }
```

Comment:
Just a minor suggestion: use `match_array` or `contain_exactly`. We can't always assume ordering by ID in Oracle.